### PR TITLE
Refactor scanner screen

### PR DESCRIPTION
**Pull request by noahsilencelab**

### CODE REVIEW

File: lib/screens/scanner/scanner_screen.dart [hunk 1]
```diff
@@ -177,70 +200,47 @@ class _ScannerScreenState extends State<ScannerScreen> {
...
-    if (isRePairingExistWallet) {
-      setState(() {
-        showAccountAlreadyPresent = true;
-        recoveryAddress = widget.repairAddress;
-      });
-      if (!widget.isRePairing) {
-        setState(() {
-          walletInfo = SupportWallet.fromJson(walletMetaData[qrMessage.walletId] ?? {});
-        });
+    if ((widget.backup != null || widget.isRePairing) && widget.recoveryWalletId != qrMessage.walletId) {
```
Comment (dakshsilence) [inline lib/screens/scanner/scanner_screen.dart]: Can use hasBackupAlready here?

File: lib/screens/scanner/scanner_screen.dart [hunk 2]
```diff
@@ -177,70 +200,47 @@ class _ScannerScreenState extends State<ScannerScreen> {
     if (_pairingState != ScannerScreenPairingState.ready) return;
 
     _updatePairingState(ScannerScreenPairingState.inProgress);
+    setState(() {
```
Comment (dakshsilence) [inline lib/screens/scanner/scanner_screen.dart:203]: We use separate function to update the values using setState, example _updatePairingState, _updateRecoveryState

Comment (noahsilencelab) [reply lib/screens/scanner/scanner_screen.dart:203]: Its seem unnecessary, this state is only set once

File: lib/screens/scanner/scanner_screen.dart (hunk 1 above)
Comment (noahsilencelab) [reply lib/screens/scanner/scanner_screen.dart]: we can

**Review by dakshsilence — APPROVED** [review]

**Pull request merged**